src/overlays/actors/ovl_En_Bal/z_en_bal.c

```diff
@@ -64,108 +149,1026 @@ static DamageTable D_80A63CE0 = {
+
+        case TEXT_STATE_DONE:
+            if (Message_ShouldAdvance(play)) {
+                EnBal_SetupGroundIdle(this);
+            }
+            break;
+
+        case TEXT_STATE_10:
+            if (Message_ShouldAdvance(play) && (this->prevTextId == 0x1D08)) {
+                this->forceEyesShut = false;
+                this->eyeTexIndex = EN_BAL_EYETEX_OPEN;
+            }
+            break;
+
+        default:
+            break;
+    }
+
+    if (this->prevTextId == 0x1D07) {
+        if ((this->skelAnime.curFrame > 29.0f) && (this->skelAnime.curFrame < 33.0f)) {
+            if (Animation_OnFrame(&this->skelAnime, 30.0f)) {
+                Actor_PlaySfx(&this->picto.actor, NA_SE_EV_CHINCLE_SPELL_EFFECT);
+            }
+            EnBal_ThrowMagicSparkles(this, play);
+        }
+        if (Animation_OnFrame(&this->skelAnime, 35.0f)) {
+            this->forceEyesShut = false;
+            this->eyeTexIndex = EN_BAL_EYETEX_OPEN;
+        }
+    }
+}
+
+void EnBal_UpdateShadow(EnBal* this) {
+    f32 homeY;
+    f32 worldY;
+    f32 shadowScale;
+
+    if ((this->actionFunc != EnBal_GroundIdle) && (this->actionFunc != EnBal_FloatIdle)) {
+        homeY = this->picto.actor.home.pos.y - this->picto.actor.floorHeight;
+        worldY = this->picto.actor.world.pos.y - this->picto.actor.floorHeight;
+        shadowScale = (homeY - worldY) / homeY;
+        shadowScale = CLAMP_MIN(shadowScale, 0.0f);
+        this->picto.actor.shape.shadowScale = 12.0f * shadowScale;
+        this->picto.actor.shape.shadowAlpha = 255.0f * shadowScale;
+    }
+}
+
+void EnBal_SetRecognizedPlayerForm(void) {
+    switch (gSaveContext.save.playerForm) {
+        case PLAYER_FORM_HUMAN:
+            SET_WEEKEVENTREG(WEEKEVENTREG_TINGLE_KNOWN_PLAYER_FORM_LOW_BIT);
+            SET_WEEKEVENTREG(WEEKEVENTREG_TINGLE_KNOWN_PLAYER_FORM_HIGH_BIT);
+            break;
+        case PLAYER_FORM_DEKU:
+            SET_WEEKEVENTREG(WEEKEVENTREG_TINGLE_KNOWN_PLAYER_FORM_LOW_BIT);
+            break;
+        case PLAYER_FORM_GORON:
+            SET_WEEKEVENTREG(WEEKEVENTREG_TINGLE_KNOWN_PLAYER_FORM_HIGH_BIT);
+            break;
+        case PLAYER_FORM_ZORA:
+            break;
+    }
+}
+
+PlayerTransformation EnBal_GetRecognizedPlayerForm(void) {
+    if (CHECK_WEEKEVENTREG(WEEKEVENTREG_TINGLE_KNOWN_PLAYER_FORM_LOW_BIT)) {
+        if (CHECK_WEEKEVENTREG(WEEKEVENTREG_TINGLE_KNOWN_PLAYER_FORM_HIGH_BIT)) {
+            return PLAYER_FORM_HUMAN;
+        }
+        return PLAYER_FORM_DEKU;
+    } else if (CHECK_WEEKEVENTREG(WEEKEVENTREG_TINGLE_KNOWN_PLAYER_FORM_HIGH_BIT)) {
+        return PLAYER_FORM_GORON;
+    }
+    return PLAYER_FORM_ZORA;
+}
+
+void EnBal_ThrowMagicSparkles(EnBal* this, PlayState* play) {
+    static Vec3f sSparkleVel = { 0.0f, 0.0f, 0.0f };
+    static Vec3f sSparkleAcc = { 0.0f, 0.0f, 0.0f };
+    static Color_RGBA8 sSparkleColorPrim = { 255, 255, 255, 255 };
+    static Color_RGBA8 sSparkleColorEnv = { 100, 200, 0, 255 };
```

Comment:
```suggestion
    static Color_RGBA8 sSparklePrimColor = { 255, 255, 255, 255 };
    static Color_RGBA8 sSparkleEnvColor = { 100, 200, 0, 255 };
```
This is the order we name these variables